### PR TITLE
Add blockly option for specifying image as workspace background

### PR DESCRIPTION
**Pull request by shakao**

### CODE REVIEW

**Review by jwunderl — APPROVED** [review]
besides spread in destructuring, LGTM!

File: core/grid.js [hunk 1]
```diff
@@ -149,8 +158,12 @@ Blockly.Grid.prototype.update = function(scale) {
   start *= scale;
   end *= scale;
 
-  this.setLineAttributes_(this.line1_, scale, start, end, half, half);
-  this.setLineAttributes_(this.line2_, scale, half, half, start, end);
+  if (this.imageOptions_) {
+    this.gridPattern_.setAttribute('patternTransform', `scale(${scale})`);
```
Comment (samelhusseini) [inline core/grid.js]: No template string in ES5.

File: core/grid.js [hunk 2]
```diff
@@ -221,6 +234,20 @@ Blockly.Grid.createDom = function(rnd, gridOptions, defs) {
...
     // x1, y1, x1, x2 properties will be set later in update.
+  } else if (gridOptions['image'] && gridOptions['image']['path']) {
+    /*
+      <pattern id="blocklyGridPattern837493" patternUnits="userSpaceOnUse"
+              width="IMAGE_WIDTH" height="IMAGE_HEIGHT">
+        <image xlink:href="IMAGE_URL" style="IMAGE_STYLE" />
+      </pattern>
+    */
+    var gridImageOptions = gridOptions['image'];
+    var { path, width, height, ...rest } = gridImageOptions;
```
Comment (samelhusseini) [inline core/grid.js]: No spread in ES5.

File: core/options.js [hunk 3]
```diff
@@ -302,9 +302,27 @@ Blockly.Options.parseGridOptions_ = function(options) {
...
+
+  // pxt-blockly: specify custom image bg for workspace
+  gridOptions.image = Blockly.Options.parseGridImageOptions_(grid);
   return gridOptions;
 };
 
+/**
+ * Parse the user-specified image options if custom image is enabled.
+ * @param {!Object} options Dictionary of options.
+ * @return {!Object} A dictionary of normalized options.
```
Comment (samelhusseini) [inline core/options.js]: ``@return {Object}`` since it is nullable.

File: core/grid.js (hunk 2 above)
Comment (samelhusseini) [reply core/grid.js]: gridImageOptions can be null. That would throw when you do ``var { ... } = null``.

**Review by samelhusseini — COMMENTED** [review]
Cool feature!

**Comment by shakao** [comment]
@samelhusseini thanks for the review! just let me know if you guys want me to make a PR or something for blockly proper

**Pull request merged**